examples/vue-code-file/eslint.config.js

```diff
@@ -1,18 +1,31 @@
-import vueParser from 'vue-eslint-parser';
+import { mergeProcessors } from 'eslint-merge-processors';
+import pluginVue from 'eslint-plugin-vue';
+import processorVueBlocks from 'eslint-processor-vue-blocks';
 import js from '@eslint/js';
 import graphqlPlugin from '@graphql-eslint/eslint-plugin';
 
 export default [
   {
-    files: ['**/*.js', '**/*.vue'],
+    files: ['**/*.js'],
     processor: graphqlPlugin.processor,
     rules: js.configs.recommended.rules,
   },
+  ...pluginVue.configs['flat/recommended'],
   {
     files: ['**/*.vue'],
-    languageOptions: {
-      parser: vueParser,
-    },
+    // `eslint-plugin-vue` will set a default processor for `.vue` files
+    // we use `eslint-merge-processors` to extend it
```

Comment:
FWIW, I don't understand this comment. What is the merge doing, and why?

Reply:
I copied it from https://github.com/antfu/eslint-processor-vue-blocks#usage

Reply:
Ah! That context helps. How about something like:
> Vue still needs to be parsed by the Vue parser for normal ESLinting. But GraphQL's lint needs to lint only the JS/TS part, so extract those as blocks using `eslint-processor-vue-blocks`. This turns the script parts of Vue SFCs into virtual JS/TS blocks inside ESLint. ESLint can then parse the JS/TS to find GraphQL parts. And finally, graphql-eslint can lint the resulting GraphQL.

Reply:
missed comment, added in https://github.com/dimaMachina/graphql-eslint/commit/e59c13fe169eb3691a5dda6e0291997ad8e3bb60 @bmulholland